designs/function-node-lifecycle/README.md

```diff
@@ -79,9 +162,38 @@ Exceptions to the initialization code are logged to console. If error handling i
 
 Export format of function node to Node-RED library currently uses comments to encode properties.  This must be extended to be able to include initialization/finilization code and required npm modules information.
 
+### Additional Properties to Function node
+
+In order to support NPM module installation and initialization/finalization code, following properties are added the Function node.
+
+| ame          | Type            | Description                        |
+| ------------ | --------------- | ---------------------------------- |
+| `initialize` | `string`        | JavaScript code for initialization |
+| `finalize`   | `string`        | JavaScript code for finalization   |
+| `modules`    | array of object | Array of NPM module specifications |
+
+- `initialize`/`finalize` property
+
+  String representation of JavaScript code executed on initialization/finalization of function node.
+
+- `modules` property
+
+  Specify NPM modules that should be installed before execution of `Function` node. It is an array of objects containing following properties:
+
+  | Name     | Type     | Description                    |
+  | -------- | -------- | ------------------------------ |
+  | `spec`   | `string` | NPM install specification      |
+  | `var`    | `string` | JavaScript variable name       |
+  | `status` | `string` | `notinstalled`/`installed`/... |
```

Comment:
I don't think the node configuration needs to know or track the status of the module. That is information the runtime would provide.

If the node's property is `notinstalled`, the runtime then installs it, the flow configuration cannot be updated by the runtime to say it is now `installed` - that can only be done when the user deploys a new flow configuration.

Reply:
removed `status` from list.